src/PowerShell/Microsoft.WinGet.Client/Helpers/AppxModuleHelper.cs

```diff
@@ -139,31 +151,79 @@ private PSObject GetAppxObject(string packageName)
                 .FirstOrDefault();
         }
 
+        private IReadOnlyList<string> GetVCLibsDependencies()
+        {
+            var vcLibsDependencies = new List<string>();
+            var vcLibsPackageObjs = this.psCmdlet.InvokeCommand
+                .InvokeScript(string.Format(GetAppxPackageByVersionCommand, VCLibsUWPDesktop, VCLibsUWPDesktopVersion));
+            if (vcLibsPackageObjs is null ||
+                vcLibsPackageObjs.Count == 0)
+            {
+                var arch = RuntimeInformation.OSArchitecture;
+                if (arch == Architecture.X64)
+                {
+                    vcLibsDependencies.Add(VCLibsUWPDesktopX64);
+                }
+                else if (arch == Architecture.X86)
+                {
+                    vcLibsDependencies.Add(VCLibsUWPDesktopX86);
+                }
+                else if (arch == Architecture.Arm64)
+                {
+                    // Deployment please figure out for me.
+                    vcLibsDependencies.Add(VCLibsUWPDesktopX64);
+                    vcLibsDependencies.Add(VCLibsUWPDesktopX86);
+                    vcLibsDependencies.Add(VCLibsUWPDesktopArm);
+                    vcLibsDependencies.Add(VCLibsUWPDesktopArm64);
+                }
+                else
+                {
+                    throw new PSNotSupportedException(arch.ToString());
+                }
+            }
+
+            return vcLibsDependencies;
+        }
+
         private void InstallVCLibsDependencies()
         {
             var vcLibsPackageObjs = this.psCmdlet.InvokeCommand
                 .InvokeScript(string.Format(GetAppxPackageByVersionCommand, VCLibsUWPDesktop, VCLibsUWPDesktopVersion));
             if (vcLibsPackageObjs is null ||
                 vcLibsPackageObjs.Count == 0)
             {
+                var packages = new List<string>();
+
                 var arch = RuntimeInformation.OSArchitecture;
-                string url;
                 if (arch == Architecture.X64)
                 {
-                    url = "https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx";
+                    packages.Add(VCLibsUWPDesktopX64);

```

Comment:
Shouldn't this just be implemented as iterating over whatever you get back from `GetVCLibsDependencies`?